core/sds-aspect-model-serializer/src/main/java/io/openmanufacturing/sds/aspectmodel/serializer/RdfModelCreatorVisitor.java

```diff
@@ -266,220 +241,292 @@ private Model createCharacteristicsModel( final Characteristic characteristic )
          model.add( resource, bamm.name(), serializePlainString( characteristic.getName() ) );
       }
 
-      characteristic.getDataType().ifPresent( type -> {
+      if ( !skipDataType && characteristic.getDataType().isPresent() ) {
+         final Type type = characteristic.getDataType().get();
          model.add( resource, bamm.dataType(), createResource( type.getUrn() ) );
-         if ( type.isComplex() ) {
-            model.add( type.accept( this, characteristic ) );
+         if ( !type.is( Scalar.class ) ) {
```

Comment:
Why don't you define the check as `type.is( Complex.class )`?

Reply:
Done